common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
+		this._setOptions(options);
+		if(this.client instanceof Realtime) {
+			this.client.connection.connectionManager.onAuthUpdated();
+		}
+	};
+
+	Auth.prototype.setClientId = function(clientId) {
+		if(clientId !== '*') {
+			if(this.clientId && this.clientId !== clientId) {
+				var msg = 'Unable to set incompatible clientId';
```

Comment:
> Why shouldn't we let people change their clientId with a reauth?

Because it is currently a fixed attribute of a connection  - you also can't resume or recover with a different `clientId`. Bad things happen to presence for example, so we don't support it.